tests/src/test/scala/org/http4s/EntityCodecSpec.scala

```diff
@@ -32,3 +32,4 @@ class EntityCodecSpec extends Http4sSpec {
 
   checkAll("EntityCodec[IO, Unit]", EntityCodecTests[IO, Unit].entityCodec)
 }
+ */
```

Comment:
@cquiroz I've lost track... do any of these new libraries floating around solve this problem?

Reply:
By adding an instance for `Eq[F[_]]`, this test can run: https://github.com/http4s/http4s/pull/4007/commits/c19cdd2093cd313ef9fb1d2e308af337a7dfba74

Do you think it's a suitable solution?

Reply:
Brief discussion on [cats-effect-dev Gitter](https://gitter.im/typelevel/cats-effect-dev?at=5fd4f24628a57c581b0bfb66).

I think this is fine for now, but I think in the long run Typelevel needs a discipline-effect built around scalacheck-effect's `PropF`.